Client/mods/deathmatch/logic/luadefs/CLuaResourceDefs.cpp

```diff
@@ -220,34 +222,24 @@ int CLuaResourceDefs::GetResourceConfig(lua_State* luaVM)
     return 1;
 }
 
-int CLuaResourceDefs::GetResourceName(lua_State* luaVM)
+std::string CLuaResourceDefs::GetResourceName(lua_State* luaVM, std::optional<CResource*> resourceElement)
 {
-    // Verify arguments
-    CResource*       pResource = NULL;
-    CScriptArgReader argStream(luaVM);
-    argStream.ReadUserData(pResource);
-
-    if (!argStream.HasErrors())
+    if (!resourceElement)
     {
-        if (pResource)
-        {
-            // Grab its name and return it
-            const char* szName = pResource->GetName();
-            if (szName)
-            {
-                lua_pushstring(luaVM, szName);
-                return 1;
-            }
-        }
-        else
-            m_pScriptDebugging->LogBadPointer(luaVM, "resource", 1);
+        CLuaMain* localVM = m_pLuaManager->GetVirtualMachine(luaVM);
+
+        if (!localVM)
+            throw std::runtime_error("Couldn't find the virtual machine");
```

Comment:
There's another problem. std::runtime_error isn't properly handled within the argument parser. That is when an expection is thrown, the game will be terminated. The argument parser is only capable to catch std::invalid_argument exception. But I don't think that it's the case when this type of expression should be used.

Reply:
I should use std::invalid_argument here then?
I have no other ideas currently.

Reply:
I would get rid of exceptions at all. Because the chance of them being thrown is negligible and doesn't depend on scripts logic.

Reply:
Okay

Reply:
> I should use std::invalid_argument here then? I have no other ideas currently. @tederis

If an error needs to be thrown, `std::invalid_argument` should be used.
ArgumentParser sees it as a custom error anyway, though it is indeed confusing.

Reply:
Then I think it's better to keep the throws with std::invalid_argument, because I don't want to return empty strings, and I don't want to return error messages in strings either, because this is not the expected operation of the function.

Reply:
It's important to note that `!localVM` is something that will almost never happen. So why then you have to print any message? For me it's better to return false in this case.

Reply:
> It's important to note that `!localVM` is something that will almost never happen. So why then you have to print any message? For me it's better to return false in this case.

Removed throws.

Reply:
If #3493 get merged you will be able to use std::logic_error instead of std::invalid_argument.